distribution/pull_v2.go

```diff
@@ -393,7 +393,7 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 	oldTagImageID, err := p.config.ReferenceStore.Get(ref)
 	if err == nil {
 		if oldTagImageID == imageID {
-			return false, nil
+			return false, addDigestReference(p.config.ReferenceStore, ref, manifestDigest, imageID)
```

Comment:
I think in some cases this can be called when we didn't download the layers or didn't verify that the layer checksums in manifest actually match with layer IDs. If we don't do that we can't trust the digest to be correct.